src/chainstate/stacks/db/blocks.rs

```diff
@@ -4746,22 +4747,70 @@ impl StacksChainState {
                     .expect("BUG: cost declined between executions");
 
                 match result {
-                    Ok((value, _, events)) => Some(StacksTransactionReceipt {
-                        transaction: TransactionOrigin::Burn(deposit_nft_op.into()),
-                        events,
-                        result: value,
-                        post_condition_aborted: false,
-                        stx_burned: 0,
-                        contract_analysis: None,
-                        execution_cost,
-                        microblock_header: None,
-                        tx_index: 0,
-                    }),
+                    Ok((value, _, mut events)) => {
+                        match &value {
+                            Value::Response(r) if r.committed == false => {
+                                // TODO: Figure out how this macro works and log some variables
+                                info!("DepositNft op returned error. Issue withdrawal tx");
```

Comment:
I think it's helpful for readability here to separate the response status check from the handling:

```suggestion
                        // check if the deposit failed, and if so, create an auto-withdraw event and add it to
                        // L1 operation's receipt.
                        let deposit_op_failed = if let Value::Response(r) = &value {
                              r.committed == false
                        };
                        if deposit_op_failed {
                             info!("DepositNft op returned error. Creating auto-withdrawal event."; "sender" => %sender, "id" => id, "asset-contract" => %subnet_contract_id);
                             
                             let nft_withdraw_event = ...;
                             events.push(nft_withdraw_event);
                        }
```

Reply:
Agreed, I will make the change

Reply:
Done. Also factored out creating the `StacksTransactionEvent` into a separate function to reduce the size of `process_deposit_nft_ops()`